net-misc/minidlna/patches/minidlna-1.2.1.patchset

```diff
@@ -0,0 +1,78 @@
+From 574cc3228dc51935c04e5b2629808af5c4f9d5a5 Mon Sep 17 00:00:00 2001
+From: begasus <begasus@gmail.com>
+Date: Sat, 28 Apr 2018 11:07:03 +0200
+Subject: Haiku doesn't have setpriority nor d_type
+
+
+diff --git a/monitor.c b/monitor.c
+index b3a17f5..2cafced 100644
+--- a/monitor.c
++++ b/monitor.c
+@@ -533,6 +533,8 @@ monitor_insert_directory(int fd, char *name, const char * path)
+ 			continue;
+ 		esc_name = escape_tag(e->d_name, 1);
+ 		snprintf(path_buf, sizeof(path_buf), "%s/%s", path, e->d_name);
++
++		#ifndef __HAIKU__
+ 		switch( e->d_type )
+ 		{
+ 			case DT_DIR:
+@@ -543,6 +545,7 @@ monitor_insert_directory(int fd, char *name, const char * path)
+ 			default:
+ 				break;
+ 		}
++		#endif
```

Comment:
With your patch, minidlna might not correctly handle nested directories, hopefully this small change would work without any ill effects.

```diff
-+		#endif
++		#else
++		type = resolve_unknown_type(path_buf, dir_types);
++		#endif
```